ctor/src/lib.rs

```diff
@@ -134,24 +135,30 @@ pub fn ctor(_attribute: TokenStream, function: TokenStream) -> TokenStream {
         // Why .CRT$XCU on Windows? https://www.cnblogs.com/sunkang/archive/2011/05/24/2055635.html
         // 'I'=C init, 'C'=C++ init, 'P'=Pre-terminators and 'T'=Terminators
 
+        let ctor_ident =
+            syn::parse_str::<syn::Ident>(format!("{}___rust_ctor___ctor", ident).as_ref())
+                .expect("Unable to create identifier");
+
         let output = quote!(
             #[cfg(not(any(target_os = "linux", target_os = "android", target_os = "freebsd", target_os = "macos", target_os = "ios", windows)))]
             compile_error!("#[ctor] is not supported on the current target");
 
+            #(#attrs)*
+            #vis #unsafety extern #abi #constness fn #ident() #block
+
             #[used]
             #[allow(non_upper_case_globals)]
             #[cfg_attr(any(target_os = "linux", target_os = "android"), link_section = ".init_array")]
             #[cfg_attr(target_os = "freebsd", link_section = ".init_array")]
             #[cfg_attr(any(target_os = "macos", target_os = "ios"), link_section = "__DATA,__mod_init_func")]
             #[cfg_attr(windows, link_section = ".CRT$XCU")]
-            #(#attrs)*
-            static #ident
+            static #ctor_ident
             :
-            #unsafety extern #abi #constness fn() =
+            unsafe extern "C" fn() =
             {
                 #[cfg_attr(any(target_os = "linux", target_os = "android"), link_section = ".text.startup")]
-                #unsafety extern #abi #constness fn #ident() #block;
-                #ident
+                unsafe extern "C" fn #ctor_ident() { #ident() };
```

Comment:
I think it might be better if we use `#ident` for the extern rather than `#ctor_ident`. We _might_ also be able to pull `#block` into here to hide it from global spot as well.